salt/modules/gpg.py

```diff
@@ -822,18 +834,40 @@ def export_key(
         salt '*' gpg.export_key keyids="['3FAD9F1E','3FBD8F1E']" user=username
 
     """
+    ret = {"res": True}
     gpg = _create_gpg(user, gnupghome)
 
     if isinstance(keyids, str):
         keyids = keyids.split(",")
 
-    if use_passphrase:
+    if secret and use_passphrase:
         gpg_passphrase = __salt__["pillar.get"]("gpg_passphrase")
         if not gpg_passphrase:
             raise SaltInvocationError("gpg_passphrase not available in pillar.")
-        ret = gpg.export_keys(keyids, secret, passphrase=gpg_passphrase)
+        result = gpg.export_keys(keyids, secret, passphrase=gpg_passphrase)
     else:
-        ret = gpg.export_keys(keyids, secret, expect_passphrase=False)
+        result = gpg.export_keys(keyids, secret, expect_passphrase=False)
+
+    if result and output:
+        with salt.utils.files.flopen(output, "w") as fout:
```

Comment:
Can we verify the file exists first before writing to it?

Reply:
```output``` is the file that will be created (the exported keys will be written this file).   
It doesn't need to exist beforehand. If it does exist, then it will be overwritten. 

That's the same behavior to you would get from something like  
```salt-call gpg.decrypt filename='/path/to/important.file.gpg' output='/tmp/decryptedfile.txt'```
```decryptedfile.txt``` doesn't need to exist beforehand. And if it does exist, it'll be overwritten.